openedx/core/lib/block_structure/block_structure.py

```diff
@@ -564,7 +564,7 @@ def remove_transformer_block_field(self, usage_key, transformer, key):
         """
         try:
             transformer_block_data = self.get_transformer_block_data(usage_key, transformer)
-            delattr(transformer_block_data, key)
+            del transformer_block_data.fields[key]
```

Comment:
For performance reasons, we chose to optimize for the path where the attribute exists - and let the try-catch handle the case where it doesn't.